File: matlab/solutions/segmentation_3_snakes_levelset/exo6.m
if do_fast
phi = phi0;
k = 0;
gW = grad(W,options);
for i=1:niter
    gD = grad(phi,options);
    d = max(eps, sqrt(sum(gD.^2,3)) );
    g = gD ./ repmat( d, [1 1 2] );
    G = W .* d .* div( g,options ) + sum(gW.*gD,3);
    phi = phi + tau*G;
    if mod(i,30)==0
        phi = perform_redistancing(phi);
    end
    if mod(i, floor(niter/4))==0
        k = k+1;
        subplot(2,2,k);
        plot_levelset(phi,0,f0);
    end
end
end

File: julia/NtSolutions/segmentation_3_snakes_levelset/exo6.jl
figure(figsize = (10, 10))
phi = copy(phi0)
k = 0
gW = NtToolBox.Grad(W, "sym", 2)

for i in 1 : niter
    gD = NtToolBox.Grad(phi, "sym", 2)
    d = max(eps().*ones(n, n), sqrt(sum(g0.^2, 3)))
    g = gD./cat(3, d[:, :], d[:, :])
    G = W.*(d.*NtToolBox.Div(g[:, :, 1], g[:, :, 2], "sym", 2)) + sum(gW.*gD, 3)
    phi = (phi + tau.*G)[:, :]
    if i % 30 == 0
        phi = perform_redistancing(phi)
    end
    if i % Base.div(niter, 4.) == 0
        k = k + 1
        subplot(2, 2, k)
        NtToolBox.plot_levelset(phi, 0, f0)
    end
end
